pkg/workceptor/kubernetes.go

```diff
@@ -399,26 +399,13 @@ func (kw *kubeUnit) runWorkUsingLogger() {
 	}
 
 	go func() {
+		var sinceTimeString string
+		var lastLineWritten string
 		for {
-			var sinceTimeString string
-
 			if errStdin != nil {
 				break
 			}
 
-			kw.pod, err = kw.clientset.CoreV1().Pods(ked.KubeNamespace).Get(kw.ctx, ked.PodName, metav1.GetOptions{})
-			if err != nil {
-				errMsg = fmt.Sprintf("Error getting pod %s/%s: %s", ked.KubeNamespace, ked.PodName, err)
-				kw.UpdateBasicStatus(WorkStateFailed, errMsg, 0)
-				logger.Error(errMsg)
-				break
-			}
-
-			if kw.pod.Status.Phase != corev1.PodRunning {
```

Comment:
I moved this stuff down below because I ran into a race condition where the pod exited quickly, but there were logs waiting to be read.